core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -266,7 +289,8 @@ private[deploy] class Worker(
             if (registerMasterFutures != null) {
               registerMasterFutures.foreach(_.cancel(true))
             }
-            val masterAddress = masterRef.address
+            val masterAddress =
+              if (preferConfiguredMasterAddress) masterAddressToConnect.get else masterRef.address
```

Comment:
Perhaps it isn't an issue but do you think we should fall back to `masterRef.address` in case `masterAddressToConnect` isn't set (instead of throwing a generic scala exception)? Something along the lines of:

```scala
val masterAddress = masterAddressToConnect match {
  case Some(master) if preferConfiguredMasterAddress => master
  case _ => masterRef.address
}
```